src/core/src/op/constant.cpp

```diff
@@ -79,6 +79,15 @@ Constant::Constant(const std::shared_ptr<ngraph::runtime::Tensor>& tensor) {
         constructor_validate_and_infer_types();
         allocate_buffer(false);
         tensor->read(get_data_ptr_nc(), tensor->get_size_in_bytes());
+        if (m_element_type == ov::element::string) {
+            // we need to re-initialize memory with separate (newly created) std::string objects with the same values
+            // because currenly it references std::string objects from input (not host) tensor
+            auto size = shape_size(m_shape);
+            auto string_ptr = static_cast<std::string*>(get_data_ptr_nc());
+            std::transform(string_ptr, string_ptr + size, string_ptr, [](std::string& value) {
```

Comment:
This is not incorrect. Especially in combination with `tensor->read` which makes raw memcpy at least in one of the derivatives -- I don't see any changes in any derivatives of runtime::Tensor::read, so I suppose that tensor->read cannot be applied for string-containing tensors.

This line looks as an attempt to copy string value to the same location, but objects weren't constructed in that location. This is not "re-initialization". Do we have a relevant test that is surprising passing?

Reply:
Honestly I couldn't find other derived from `ngraph::runtime::Tensor` classes except `HostTensor`.